server/routes/nextPayment.routes.js

```diff
@@ -50,4 +50,33 @@ router.get('/all', authenticate, isAdminOrUserOrVendor, NextPaymentController.ge
  */
 router.get('/reminder', NextPaymentController.sendReminder);
 
+/**
+ * @swagger
+ * /next-payment/calculate/:id:
+ *   get:
+ *     tags:
+ *       - NextPayment
+ *     description: Recalculate next payment for an offer
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/NextPayment'
+ *      description: Recalculate next payment for an offer
+ *     responses:
+ *      '200':
+ *        description: Next payment recalculated
+ *      '500':
+ *       description: Internal server error
+ */
+router.get(
+  '/calculate/:id',
+  hasValidObjectId,
+  authenticate,
+  isAdminOrUserOrVendor,
```

Comment:
Lets make this an admin only thing 🙏🏽 
```suggestion
  isAdmin,
```